src/cider/nrepl/middleware/out.clj

```diff
@@ -0,0 +1,86 @@
+(ns cider.nrepl.middleware.out
+  "Change *out* to print on connections in addition to process out.
+
+  Automatically changes the root binding of *out* to print to any
```

Comment:
Won't it be simpler to just send some message from the sessions we care about when we create them (e.g. `:init-output`). It might be nice to have a way to "unsubscribe" as well.